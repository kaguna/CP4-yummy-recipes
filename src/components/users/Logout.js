/**
 * This destroys the user session.
 */
import React, { Component } from 'react';

class Logout extends Component {
    logoutHandler = () => {
      window.localStorage.clear();
      window.location.reload();
      this.props.history.push('./Login');
    };
    componentDidMount() {
      this.logoutHandler();
    }
    render() {
      return (
        <span>
                logging out...
        </span>
      );
    }
}
export default Logout;
