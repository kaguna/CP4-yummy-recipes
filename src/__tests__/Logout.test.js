import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Logout from '../components/users/Logout';

describe('Logout component', () => {
  const wrapper = shallow(<Logout />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
