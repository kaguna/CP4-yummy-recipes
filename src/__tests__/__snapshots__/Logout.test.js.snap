// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component renders properly 1`] = `
<span>
  logging out...
</span>
`;
